### PR TITLE
CLN: remove unnecessary Series._convert calls

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @WillAyd my confidence on the correctness here is only "pretty sure", so could use an extra set of eyes.

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
Definitely a more logical way to do things

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1342,14 +1342,10 @@ def first_not_none(values):
 
             # values are not series or array-like but scalars
             else:
-                # only coerce dates if we find at least 1 datetime
-                should_coerce = any(isinstance(x, Timestamp) for x in values)
```
Comment (jorisvandenbossche) [inline pandas/core/groupby/generic.py]: Does this change behaviour?

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -386,7 +386,7 @@ def _wrap_aggregated_output(
         result = self._wrap_series_output(
             output=output, index=self.grouper.result_index
         )
-        return self._reindex_output(result)._convert(datetime=True)
+        return self._reindex_output(result)
```
Comment (jorisvandenbossche) [inline pandas/core/groupby/generic.py]: Do you have an explanation of why this is not needed anymore?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]
this is all fine. these were put in years ago when we didn't coerce back to datetimes for example oftentimes.

**Comment by jreback** [comment]
@jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
Brock, can you respond to my comments?

File: pandas/core/groupby/generic.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: No, just an observation that we dont have any tests that break without it.

**Comment by jreback** [comment]
can you merge master, this looks ok. I guess @jorisvandenbossche has some comments

**Comment by jreback** [comment]
this is fine

**Pull request merged**